packages/react/src/internal/__tests__/useResizeObserver-test.js

```diff
@@ -5,96 +5,77 @@
-import { render, screen } from '@testing-library/react';
+import React, { useRef } from 'react';
+import { render, screen, act } from '@testing-library/react';
 import { useResizeObserver } from '../useResizeObserver';
-import { mockHTMLElement } from '../__mocks__/mockHTMLElement';
-
-/**
- * @param {number} base - Represents the initial size used for the resizing example
- */
-const sizes = (base) => ({
-  offsetWidth: {
-    ['observed-el']: base,
-  },
-  offsetHeight: { ['observed-el']: base / 2 },
-});
-
-/**
- * @param {HTMLElement} el - The element being mocked for resizing functionality
- * @param {string} property - HTML property being mocked (in this scenario, these properties come from sizes())
- */
-const testSizes = (el, property) => {
-  const classes = el.getAttribute('class').split(' ');
-  const container = el.closest('.test-container');
-
-  const base = container ? parseInt(container.style.width, 10) : 400;
-  const propSizes = sizes(base)[property];
-
-  if (propSizes) {
-    for (let cls of classes) {
-      const val = propSizes[cls] ? propSizes[cls] : -1;
-      if (val >= 0) {
-        return val;
-      }
-    }
-  }
-
-  // The test should never get here as all cases should be catered for in setup.
-  return base;
-};
-
-const mockSizes = () => {
-  const mocks = {};
-
-  const keys = Object.keys(sizes(-1));
-  for (let i = 0; i < keys.length; i++) {
-    mocks[keys[i]] = {
-      get: function () {
-        return testSizes(this, keys[i]);
-      },
-    };
-  }
-
-  return mocks;
-};
-
-const TestComponent = () => {
-  const observedElement = useRef(null);
-  const { width, height } = useResizeObserver({ ref: observedElement });
-  const [content, setContent] = useState(-1);
-
-  useEffect(() => {
-    setContent(`width: ${width}, height: ${height}`);
-  }, [height, width]);
 
+const ResizeTest = ({ onResize = null }) => {
+  const ref = useRef(null);
+  const { width, height } = useResizeObserver({ ref, onResize });
   return (
-    <div ref={observedElement} className="observed-el">
-      {content}
+    <div ref={ref} data-testid="observed-element">
+      width: {width}, height: {height}
     </div>
   );
 };
 
+const defaultSize = 100;
 describe('useResizeObserver', () => {
-  const { ResizeObserver } = window;
-  let mockElement;
+  let savedObserverCb;
 
   beforeEach(() => {
-    mockElement = mockHTMLElement(mockSizes());
-    window.ResizeObserver = jest.fn().mockImplementation(() => ({
-      observe: jest.fn(),
-      unobserve: jest.fn(),
-      disconnect: jest.fn(),
-    }));
+    window.ResizeObserver = jest.fn().mockImplementation((cb) => {
+      savedObserverCb = cb;
+      return {
+        observe: jest.fn(),
+        unobserve: jest.fn(),
+        disconnect: jest.fn(),
+      };
+    });
   });
 
   afterEach(() => {
-    mockElement.mockRestore();
-    window.ResizeObserver = ResizeObserver;
+    jest.clearAllMocks();
   });
 
+  // Triggers resize from the saved resize observer callback
```

Comment:
This comment should be a JSDoc comment. It's describing a function.

```suggestion
  /** Triggers resize from the saved resize observer callback */
```

Reply:
Addressed via https://github.com/carbon-design-system/carbon/pull/19452/commits/6633de6ef908dc44c4fac5ba21ec94e8a931a83c